parser/src/test/java/io/specmesh/apiparser/AsyncApiParserTest.java

```diff
@@ -116,4 +121,75 @@ public void shouldReturnProducerMessageSchema() {
                 channelsMap.values().iterator().next().publish().message().tags(),
                 Matchers.hasSize(2));
     }
+
+    @Test
+    public void shouldSupportOldSchemaRefLookup() {
+        final var channels = API_SPEC_KEY_VALUE.channels();
+
+        final var oldStyleChannel = channels.get("simple.schema-demo._public.example_old");
+        oldStyleChannel.publish().message().schemaRef();
+
+        try {
+            oldStyleChannel.publish().message().keyType();
+            fail("should have failed on KeyLookup");
+        } catch (Exception ex) {
+
+        }
+        // throw not supports
+        try {
+            oldStyleChannel.publish().message().valueType();
```

Comment:
Why couldn't this return "simple_schema_demo_user-signedup.avsc"?

Reply:
true - ill change it